# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/yajl/hotfix-cmake-4.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- yajl-fee1ebef9fa7dc0e9f5a23f37123b19b68c796ff/CMakeLists.txt.vanilla	2025-06-29 15:41:45.578360984 +0200
+++ yajl-fee1ebef9fa7dc0e9f5a23f37123b19b68c796ff/CMakeLists.txt	2025-06-29 15:41:57.963410794 +0200
@@ -12,7 +12,7 @@
 # ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 # OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0...3.10)
 
 PROJECT(YetAnotherJSONParser C)
 
--- yajl-fee1ebef9fa7dc0e9f5a23f37123b19b68c796ff/verify/CMakeLists.txt.vanilla	2025-06-29 15:42:36.803736888 +0200
+++ yajl-fee1ebef9fa7dc0e9f5a23f37123b19b68c796ff/verify/CMakeLists.txt	2025-06-29 15:46:21.772140555 +0200
@@ -28,11 +28,8 @@
 
 TARGET_LINK_LIBRARIES(json_verify yajl_s)
 
-# copy in the binary
-GET_TARGET_PROPERTY(binPath json_verify LOCATION)
-
 ADD_CUSTOM_COMMAND(TARGET json_verify POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different ${binPath} ${binDir})
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:json_verify> ${binDir})
 
 IF (NOT WIN32)
   INSTALL(TARGETS json_verify RUNTIME DESTINATION bin)
--- yajl-fee1ebef9fa7dc0e9f5a23f37123b19b68c796ff/reformatter/CMakeLists.txt.vanilla	2025-06-29 15:42:10.361454060 +0200
+++ yajl-fee1ebef9fa7dc0e9f5a23f37123b19b68c796ff/reformatter/CMakeLists.txt	2025-06-29 15:46:14.131358611 +0200
@@ -28,11 +28,8 @@
 
 TARGET_LINK_LIBRARIES(json_reformat yajl_s)
 
-# copy the binary into the output directory
-GET_TARGET_PROPERTY(binPath json_reformat LOCATION)
-
 ADD_CUSTOM_COMMAND(TARGET json_reformat POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different ${binPath} ${binDir})
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:json_reformat> ${binDir})
 
 IF (NOT WIN32)
   INSTALL(TARGETS json_reformat RUNTIME DESTINATION bin)
